app/questionnaire/router.py

```diff
@@ -175,11 +202,59 @@ def is_survey_complete(self):
 
         return True
 
+    def is_path_complete(self, path):
+        location = self._get_first_incomplete_location(path)
+        if not location or (
+            location == path[-1]
+            and self._schema.get_block(location.block_id)['type'] == 'SectionSummary'
+        ):
+            return True
+
+        return False
+
     def get_section_return_location_when_section_complete(
         self, routing_path
     ) -> Location:
         return self._get_location_of_section_summary(routing_path) or routing_path[0]
 
+    def full_routing_path(self):
+        path = []
+        for section_id in self.enabled_section_ids:
+
+            repeating_list = self._schema.get_repeating_list_for_section(section_id)
+
+            if repeating_list:
+                for list_item_id in self._list_store[repeating_list].items:
+                    path = path + list(
+                        self._path_finder.routing_path(
+                            section_id=section_id, list_item_id=list_item_id
+                        )
+                    )
+            else:
+                path = path + list(
+                    self._path_finder.routing_path(section_id=section_id)
+                )
+
+        return path
+
+    def _get_first_incomplete_location(self, path):
+        for location in path:
+            block = self._schema.get_block(location.block_id)
+            block_type = block.get('type')
+
+            is_location_complete = (
```

Comment:
Not your code, but:

```
completed_block_ids = self._progress_store.get_completed_block_ids(
    section_id=location.section_id, list_item_id=location.list_item_id
)
is_location_complete = location.block_id in completed_block_ids
```

Reads easier I think :)

Reply:
Should you not be 🛌  😁. But since this is used in multiple places I made a method called `is_location_complete` instead.